### DISCUSSION TITLE
How prevent closing a menu after clicking on  menu item with checkbox

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#11695] [Q&A] [answered]
I have a menu item with a checkbox:

        <MenuItem Header="Checked MenuItem">
          <MenuItem.Icon>
              <CheckBox IsChecked="True" ClipToBounds="False"/>
          </MenuItem.Icon>
        </MenuItem>

The menu stays open after clicking on the checkbox, which is what I want.
However, the menu closes after clicking  elsewhere on the menu item.

How can this be prevented?

**Comment by timunie** [comment]
Can you listen to `Menu.Closing`-event and set `e.Cancel=true`? Can't test it rn, but may work.

**Comment by wgraaf** [comment, marked as answer]
I ended up putting a button in the menu item header:

        <MenuItem>
          <MenuItem.Icon>
            <CheckBox IsChecked="{Binding CheckedMenuItem, Mode=TwoWay}" ClipToBounds="False"/>
          </MenuItem.Icon>
          <MenuItem.Header>
            <Button Content="CheckedMenuItem" 
               HorizontalContentAlignment="Left" Padding="0" Margin="0" MinHeight="24" BorderThickness="0" 
              Background="Transparent" 
              Command="{Binding ToggleCheckedMenuItemCommand}"/>
          </MenuItem.Header>
        </MenuItem>
